internal/app/pactproxy/response_modification_writer.go

```diff
@@ -7,36 +7,46 @@ import (
 )
 
 type ResponseModificationWriter struct {
-	res          http.ResponseWriter
-	interactions []*interaction
-	statusCode   int
+	res              http.ResponseWriter
+	interactions     []*interaction
+	originalResponse []byte
+	statusCode       int
 }
 
 func (m *ResponseModificationWriter) Header() http.Header {
 	return m.res.Header()
 }
 
-func (m *ResponseModificationWriter) Write(b []byte) (written int, err error) {
-	written = len(b)
+func (m *ResponseModificationWriter) Write(b []byte) (int, error) {
+	originalResponseLength, err := strconv.Atoi(m.res.Header().Get("Content-Length"))
+	if err != nil {
+		return 0, err
+	}
+
+	m.originalResponse = append(m.originalResponse, b...)
+	if len(m.originalResponse) != originalResponseLength {
+		return len(b), nil
+	}
 
+	var modifiedBody []byte
 	for _, i := range m.interactions {
-		b, err = i.Modifiers.modifyBody(b)
+		modifiedBody, err = i.Modifiers.modifyBody(m.originalResponse)
```

Comment:
~~I suspect this will only work when we call it the final time, as the content won't all be present otherwise.~~ just realised you're returning early above when this isn't the case, nice work!

Nonetheless could we confirm with a test with a large body and a modifier?

Reply:
Done. Tests are added to verify body and status code modifiers.